plugins/workload-management/src/main/java/org/opensearch/plugin/wlm/action/CreateQueryGroupRequest.java

```diff
@@ -0,0 +1,243 @@
+        this.resiliencyMode = mode;
+        this.updatedAtInMillis = updatedAtInMillis;
+    }
+
+    /**
+     * Constructor for CreateQueryGroupRequest
+     * @param in - A {@link StreamInput} object
+     */
+    public CreateQueryGroupRequest(StreamInput in) throws IOException {
+        super(in);
+        name = in.readString();
+        _id = in.readString();
+        resiliencyMode = ResiliencyMode.fromName(in.readString());
+        resourceLimits = in.readMap((i) -> ResourceType.fromName(i.readString()), StreamInput::readGenericValue);
+        updatedAtInMillis = in.readLong();
+    }
+
+    @Override
+    public CreateQueryGroupRequest read(StreamInput in) throws IOException {
+        return new CreateQueryGroupRequest(in);
+    }
+
+    /**
+     * Generate a CreateQueryGroupRequest from XContent
+     * @param parser - A {@link XContentParser} object
+     */
+    public static CreateQueryGroupRequest fromXContent(XContentParser parser) throws IOException {
+
+        while (parser.currentToken() != XContentParser.Token.START_OBJECT) {
+            parser.nextToken();
+        }
+
+        if (parser.currentToken() != XContentParser.Token.START_OBJECT) {
+            throw new XContentParseException("expected start object but got a " + parser.currentToken());
+        }
+
+        XContentParser.Token token;
+        String fieldName = "";
+        String name = null;
+        ResiliencyMode mode = null;
+
+        // Map to hold resources
+        final Map<ResourceType, Object> resourceLimits = new HashMap<>();
+        while ((token = parser.nextToken()) != null) {
+            if (token == XContentParser.Token.FIELD_NAME) {
+                fieldName = parser.currentName();
+            } else if (token.isValue()) {
+                if (fieldName.equals("name")) {
+                    name = parser.text();
+                } else if (fieldName.equals("resiliency_mode")) {
+                    mode = ResiliencyMode.fromName(parser.text());
+                } else {
+                    throw new XContentParseException("unrecognised [field=" + fieldName + " in QueryGroup");
+                }
+            } else if (token == XContentParser.Token.START_OBJECT) {
+                if (!fieldName.equals("resourceLimits")) {
+                    throw new XContentParseException("Invalid field passed. QueryGroup does not support " + fieldName + ".");
+                }
+                while ((token = parser.nextToken()) != XContentParser.Token.END_OBJECT) {
+                    if (token == XContentParser.Token.FIELD_NAME) {
+                        fieldName = parser.currentName();
+                    } else {
+                        resourceLimits.put(ResourceType.fromName(fieldName), parser.doubleValue());
+                    }
+                }
+            }
+        }
+        return new CreateQueryGroupRequest(name, UUIDs.randomBase64UUID(), mode, resourceLimits, Instant.now().getMillis());
```

Comment:
We can initialize `UUIDs.randomBase64UUID()` and `Instant.now().getMillis()` within the constructor, to not pass as parameter, as convenient initialization option for first time creation?